executor/executor.go

```diff
@@ -183,21 +183,25 @@ func handleBuildTimeout(f *os.File, timeoutErr error) {
 	f.Write([]byte{4})
 }
 
-func filterTeardowns(build screwdriver.Build) ([]screwdriver.CommandDef, []screwdriver.CommandDef) {
+func filterTeardowns(build screwdriver.Build) ([]screwdriver.CommandDef, []screwdriver.CommandDef, []screwdriver.CommandDef) {
 	userCommands := []screwdriver.CommandDef{}
-	teardownCommands := []screwdriver.CommandDef{}
+	sdTeardownCommands := []screwdriver.CommandDef{}
+	userTeardownCommands := []screwdriver.CommandDef{}
 
 	for _, cmd := range build.Commands {
-		isTeardown, _ := regexp.MatchString("sd-teardown-*", cmd.Name)
+		isSdTeardown, _ := regexp.MatchString("sd-teardown-*", cmd.Name)
+		isUserTeardown, _ := regexp.MatchString("teardown-*", cmd.Name)
```

Comment:
I think this regexp will overlap with `sd-teardown-*`, since the MatchString 
You need to use `^sd-teardown-*` and `^teardown-*` to match the beginning of the string

Reply:
Thank you. I'll fix it